client/src/components/profile/components/time-line.tsx

```diff
@@ -268,11 +269,16 @@ function useIdToNameMap(): Map<string, NameMap> {
           // @ts-expect-error Graphql needs typing
           title,
           // @ts-expect-error Graphql needs typing
-          fields: { slug }
+          fields: { slug, blockName }
         }
       }
     }) => {
-      idToNameMap.set(id, { challengeTitle: title, challengePath: slug });
+      idToNameMap.set(id, {
+        challengeTitle: `${blockName} ${
+          title.includes('Step') ? `- ${title}` : ''
+        }`,
```

Comment:
Okay, I have an idea:

I can change the current logic to-
```
${title.includes("Step") ? `${blockName} - ` : "" }${title}
```
so that it always shows the challengeTitle?

**OR**

Unless it'd be better to create another string in:
```
interface NameMap {
  challengeTitle: string;
  challengePath: string;
}
```
like `challengeBlockName : string;` , set that to `${getTitleFromId(id)} Certification` in the Map constructor `idToNameMap` , then have `blockName` filtered out only the block names from step-based projects by challengeType then update the current logic with that?

Reply:
The first approach should be fine.